test/authenticate.test.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const http = require('http')
+const test = require('tap').test
+const websocket = require('websocket-stream')
+const proxyquire = require('proxyquire')
+let server, cliToken
+let simulateTimeout = false
+let simulateNoToken = false
+
+test('Before all', function (t) {
+  server = http.createServer(() => {})
+
+  websocket.createServer({ server }, conn => {
+    conn.on('data', token => {
+      cliToken = token.toString('utf8')
+      if (simulateTimeout) {
+        conn.write('timeout\n')
+      } else if (simulateNoToken) {
+        conn.write('\n')
+      } else {
+        conn.write('jwtToken\n')
+      }
+      conn.end()
+    })
+  })
+
+  server.listen(0, function () {
+    t.plan(1)
+    t.ok(server)
+  })
+})
+
+test('authenticate', function (t) {
```

Comment:
this is probably better be an async function, so that we can use await afterwards.

Reply:
refactored